### ISSUE TITLE
Reconnect agents based on version in the TCP balancing system

### ISSUE THREAD

**Issue by Selutario** [wazuh/wazuh#14370]
## Description

The balancing system developed at https://github.com/wazuh/wazuh/issues/13195 is based on sending reconnection requests to the agents. However, this feature (forcing reconnection) was introduced in v4.3.0, so it is not possible to balance those agents with lower versions using this method.

Therefore, we should apply `version>"Wazuh v4.3.0"` filter when obtaining the ID of the agents to reconnect:
https://github.com/wazuh/wazuh/blob/1ade8b4d0593637c00e3185cb25bfdf485469765/framework/wazuh/core/cluster/agents_reconnect.py#L269-L274

In case fewer agents than expected are obtained (when there are too many old agents), a warning should be logged saying that the balancing will not work correctly for the reason described above. Other additional measures may also need to be taken in such cases (for example stopping the system). 

The goal is to avoid the system from stopping as if the balancer were misconfigured (agents not being rebalanced) since the requests would only be sent to those that can reconnect.

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [ ] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [ ] Core (`framework/wazuh/core/tests/`)
  - [ ] SDK (`framework/wazuh/tests/`)
  - [ ] RBAC (`framework/wazuh/rbac/tests/`)
  - [ ] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [ ] Affected tests 
  - [ ] Affected RBAC (black and white) tests
- [ ] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [ ] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [ ] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [ ] Changelog (`CHANGELOG.md`)

### wazuh/wazuh-documentation
- [ ] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [ ] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

**Comment by AdriiiPRodri** [comment]
## Status update

At the moment we have decided to add a comment indicating that this functionality will only be effective for agents whose version is equal or higher than 4.3.0 and therefore, agents with lower version will be ignored.

In this way we can avoid the need to add new procedures. If we apply the version filter in the `get_agents` function of agents_reconnect as indicated in the initial comment of the issue we could stop the system by changes in the agents status.

This happens because before reaching `get_agents`, we get the agents that need rebalancing in a different function, so we cannot simply check if the number of agents obtained in `get_agents` with the filter is the same as the one obtained in `need_balance` without filter:

https://github.com/wazuh/wazuh/blob/1ade8b4d0593637c00e3185cb25bfdf485469765/framework/wazuh/core/cluster/agents_reconnect.py#L218-L252

To summarize, as a first approximation we will consider only the agents whose version is 4.3.0 or higher for the cluster balancing, in this way we will not add unnecessary load because it would be adding code which would have an expiration date since normally the Wazuh installations are updated.

**Comment by davidjiglesias** [comment]
We need to further investigate the solution approach for this
